src/env.h

```diff
@@ -266,6 +266,13 @@ struct node_ares_task {
   RB_ENTRY(node_ares_task) node;
 };
 
+// 4096 bytes on x64 and 4088 on ia32.
+struct node_destroy_ids_list {
+  size_t length;
+  node_destroy_ids_list* prev;
+  int64_t ids[510];
+};
```

Comment:
If I judge the implementation complexity between using and not using the STL to be comparable then I usually choose to not use the STL. Though the thought of using std::vector doesn't bother me. Can switch it if that's your preference.